engine-tests/src/test/java/org/terasology/engine/rendering/assets/mesh/VertexGLAttributeTest.java

```diff
@@ -0,0 +1,259 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.engine.rendering.assets.mesh;
+
+import org.joml.Vector2f;
+import org.joml.Vector2fc;
+import org.joml.Vector3f;
+import org.joml.Vector3fc;
+import org.joml.Vector4f;
+import org.joml.Vector4fc;
+import org.junit.Test;
+import org.terasology.engine.rendering.assets.mesh.resource.GLAttributes;
+import org.terasology.engine.rendering.assets.mesh.resource.VertexAttributeBinding;
+import org.terasology.engine.rendering.assets.mesh.resource.VertexFloatAttributeBinding;
+import org.terasology.engine.rendering.assets.mesh.resource.VertexIntegerAttributeBinding;
+import org.terasology.engine.rendering.assets.mesh.resource.VertexResource;
+import org.terasology.engine.rendering.assets.mesh.resource.VertexResourceBuilder;
+import org.terasology.nui.Color;
+import org.terasology.nui.Colorc;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class VertexGLAttributeTest {
+
+    @Test
+    public void testFloatBinding() {
+        VertexResourceBuilder builder = new VertexResourceBuilder();
+        VertexFloatAttributeBinding a1 = builder.add(0, GLAttributes.FLOAT_1_VERTEX_ATTRIBUTE);
+        VertexResource resource = builder.build();
+
+        a1.put(10.0f);
+        a1.put(15.0f);
+        a1.put(15.5f);
+        a1.put(2.0f);
+        a1.put(-100.0f);
```

Comment:
Fun ways to test floating-point data structures include: `-0` (negative zero), `NaN`, [`POSITIVE_INFINITY`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Float.html#POSITIVE_INFINITY).

(This may or may not actually be relevant for this application.)